### DISCUSSION TITLE
New Install Failing - Salt Repo Error

### DISCUSSION THREAD

**Discussion by imalc3142** [Security-Onion-Solutions/securityonion#4934] [Unsupported Versions] [answered]
Attempting to do a new install on Ubuntu 18.04 and am getting the following logged to sosetup.log after the 
system checks:

Executing command with retry support: apt-get update
E: The repository 'http://archive.repo.saltstack.com/py3/ubuntu/18.04/amd64/archive/3003 bionic Release' does not have a Release file.
Results: Hit:2 https://packages.wazuh.com/3.x/apt stable InRelease
Hit:3 https://download.docker.com/linux/ubuntu bionic InRelease
Hit:4 http://us.archive.ubuntu.com/ubuntu bionic InRelease
Hit:5 http://us.archive.ubuntu.com/ubuntu bionic-updates InRelease
Hit:6 http://us.archive.ubuntu.com/ubuntu bionic-backports InRelease
Hit:7 http://us.archive.ubuntu.com/ubuntu bionic-security InRelease
Ign:1 https://archive.repo.saltproject.io/py3/ubuntu/18.04/amd64/archive/3003 bionic InRelease
Err:8 https://archive.repo.saltproject.io/py3/ubuntu/18.04/amd64/archive/3003 bionic Release
  404  Not Found [IP: 65.8.242.63 443]
Reading package lists... (100)
Command failed with exit code 100; will retry in 10 seconds (1 / 50)...

Then the script fails after 50 attempts.

sosetup.log attached
[sosetup.log](https://github.com/Security-Onion-Solutions/securityonion/files/6879444/sosetup.log)

Is this anything you can fix? or am I doing something wrong.

Thanks,

Al

**Comment by ghost** [comment]
I'm having a similar issue, except instead of a 404 error, I'm getting 500 Bad redirection.

**Comment by defensivedepth** [comment]
@imalc3142 Do you have a content filter or anything else that might be messing with the connection?

**Comment by imalc3142** [comment]
Hi Josh,

I do not have any filters in place.  The issue is either with a missing
repository on Salt's end or the install script is incorrect.

The script is looking for:
http://archive.repo.saltstack.com/py3/ubuntu/18.04/amd64/archive/3003
bionic Release

If you go to:
http://archive.repo.saltstack.com/py3/ubuntu/18.04/amd64/archive/

you will see that there is not a /3003 directory.

This might be on Salt's end, but can someone at Security Onion's side take
a look?

Thanks,

Al Cooper  CISSP
HMC Networks, Inc.
719.390.8078

**Reply by defensivedepth** [reply]
There have been multiple 1804 installs done internally over the last couple days and had no issues.

Looking at your log, the failure is from when setup runs `apt-get update`

If you run that manually (outside of the setup script), do you see failures?

**Comment by imalc3142** [comment]
Hi Josh,

***@***.***:~# sudo apt update
Hit:1 https://download.docker.com/linux/ubuntu bionic InRelease
Hit:3 http://us.archive.ubuntu.com/ubuntu bionic InRelease

Hit:4 https://packages.wazuh.com/3.x/apt stable InRelease

Get:5 http://us.archive.ubuntu.com/ubuntu bionic-updates InRelease [88.7
kB]
Get:6 http://us.archive.ubuntu.com/ubuntu bionic-backports InRelease [74.6
kB]
Get:7 http://us.archive.ubuntu.com/ubuntu bionic-security InRelease [88.7
kB]
Get:8 http://us.archive.ubuntu.com/ubuntu bionic-updates/main amd64
Packages [2161 kB]
Ign:2
https://archive.repo.saltproject.io/py3/ubuntu/18.04/amd64/archive/3003
bionic InRelease
Err:9
https://archive.repo.saltproject.io/py3/ubuntu/18.04/amd64/archive/3003
bionic Release
  404  Not Found [IP: 65.8.242.67 443]
Reading package lists... Done

if you go to is a web browser:
https://archive.repo.saltproject.io/py3/ubuntu/18.04/amd64/archive/
Do you have a 3003/ directory?  I'm do not, from two different networks.

And I am running Ubuntu 18.04

***@***.***:~# lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 18.04.5 LTS
Release:        18.04
Codename:       bionic

Al Cooper  CISSP
HMC Networks, Inc.
719.390.8078

**Reply by defensivedepth** [reply]
Can you `ls /etc/yum.repos.d` ?

**Comment by imalc3142** [comment]
I'm running Ubuntu, not Redhat so I am not using yum.

Here is my sources.list.d

***@***.***:/# ll /etc/apt/sources.list.d
total 20
drwxr-xr-x 2 root root 4096 Jul 20 21:15 ./
drwxr-xr-x 7 root root 4096 Jul 20 20:39 ../
-rw-r--r-- 1 root root   86 Jul 20 21:15 saltstack.list
-rw-r--r-- 1 root root   78 Jul 20 20:38 saltstack.list.save
-rw-r--r-- 1 root root   52 Jul 20 20:39 wazuh.list

Al Cooper  CISSP
HMC Networks, Inc.
719.390.8078

**Comment by defensivedepth** [comment]
Not sure why the archive one is there. Might try to grep archive in the sources.list directory/file(s) and remark/remove them, or do an apt clean.

**Comment by imalc3142** [comment, marked as answer]
I think that fixed it.  The server is running, what looks like, a normal
install.

Thanks for your help!!

Al Cooper  CISSP
HMC Networks, Inc.
719.390.8078

**Reply by defensivedepth** [reply]
Great!
